checks/dangerous_workflow.go

```diff
@@ -120,23 +128,70 @@ func validateGitHubActionWorkflowPatterns(path string, content []byte, dl checke
 		return false, err
 	}
 
+	// 3. Check for secrets used in workflows triggered by pull requests.
+	if err := validateSecretsInPullRequests(workflow, path, dl, pdata); err != nil {
+		return false, err
+	}
+
 	// TODO: Check other dangerous patterns.
 	return true, nil
 }
 
+func validateSecretsInPullRequests(workflow *actionlint.Workflow, path string,
+	dl checker.DetailLogger, pdata *patternCbData) error {
+	// We need pull request trigger.
+	if !usesPullRequestTrigger(workflow) {
+		return nil
+	}
+
+	// We need code checkout.
+	if !workflowUsesCodeCheckout(workflow) {
+		return nil
+	}
+
+	// Secrets used in env at the top of the wokflow.
+	if err := checkSecretInEnv(workflow.Env, path, dl, pdata); err != nil {
+		return err
+	}
+
+	// Secrets used on jobs.
+	for _, job := range workflow.Jobs {
```

Comment:
correct, the check is in `checkJobForUsedSecrets()`

Note: I'm going to update the code today to also verify whether an `environment` is declared or not, since this indicates a env secret and is typically gated by an approval, so we can discard such jobs. We can add further logic later to read the actual environment rules if we want

Reply:
updated the PR description to reflect my last comment.